client/src/components/app/App.js

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { Provider } from "react-redux";
+import { Redirect, Route, Switch } from "react-router-dom";
+
+import AllArticlesContainer from "components/article/AllArticlesContainer";
+import DetailedArticle from "components/article/DetailedArticle";
+import Login from "components/user/Login";
+import Logout from "components/user/Logout";
+import NoMatch from "./NoMatch";
+import Signup from "components/user/Signup";
+import TopBar from "components/app/TopBar";
+import User from "components/user/User";
+
+import store from "redux/store";
+import LocalStorage from "LocalStorage";
+
+class App extends React.Component {
+  state = {
+    loggedInUser: {},
+  };
+
+  constructor(props) {
+    super(props);
+    LocalStorage.subscribe(this.localStorageUpdated);
+  }
+
+  componentDidMount() {
+    this.localStorageUpdated();
+  }
+
+  localStorageUpdated = () => {
+    const loggedInUser = LocalStorage.getLoggedInUser();
+    this.setState({ loggedInUser });
+  };
+
+  isLoggedIn = () => {
+    return Object.keys(this.state.loggedInUser).length !== 0;
+  };
+
+  render() {
+    return (
+      <div>
+        <Route
```

Comment:
If this is a header or some sort of components that will be rendered with all routes then it should NOT be wrapped within a `Rout` component

Reply:
The problem is that inside TopBar I need the current `pathname` to be able to avoid rendering the login/signup buttons when on login/signup pages.

Router already sends you the current location and every time it changes the props will change, I tried to find a way to subscribe to changes in `window.location` but couldn't find a working solution.

What would be the alternative to subscribe to current `pathname` if I don't use the router?

Reply:
Update: Found `withRouter` in react-router that can send all the props I need without adding a specific `Route` to the component